### DISCUSSION TITLE
Problem to use external database

### DISCUSSION THREAD

**Discussion by valodo** [osm-search/Nominatim#2442] [Q&A]
Good morning,
I am new in Nominatim, I would like to use a docker container where is installed Nominatim and link it to an external postgres database where data are already imported.

I use this repository https://github.com/mediagis/nominatim-docker/tree/master/3.7 as reference for Nominatim docker container and I add to .env file under the project directory the variables NOMINATIM_DATABASE_MODULE_PATH and NOMINATIM_DATABASE_DNS as written in https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#importing-nominatim-to-an-external-postgresql-database. However, when I run nominatim refresh --functions and nominatim admin --check-database, I receive the following error:

`2021-09-06 07:38:08: Using project directory: /nominatim
2021-09-06 07:38:08: Create functions
2021-09-06 07:38:08: FATAL: Cannot connect to database: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?`

I am able to ping the database server from the docker container and I am also able to connect to the database server from docker container by using the command line to connect to remote postgres database. 

Can you help me?

Thank you in advance.

**Comment by lonvia** [comment]
The error message indicates that Nominatim still tries to connect via a socket. That means that your connection parameters are not recognised. Might be a simple typo. You write about `NOMINATIM_DATABASE_DNS` up there, but the parameter is called `NOMINATIM_DATABASE_DSN`.
